### ISSUE TITLE
The allow_output_check argument is no longer available for virttest calls

### ISSUE THREAD

**Issue by pevogam** [avocado-framework/avocado-vt#3299]
The recent changes in [beee20f45cbe63b9a6c70a344c1b13097243151d](https://github.com/avocado-framework/avocado/commit/beee20f45cbe63b9a6c70a344c1b13097243151d) caused:
```
[stderr] Traceback (most recent call last):
[stderr]   File "/usr/lib/python3.9/site-packages/avocado_vt/plugins/vt_runner.py", line 52, in runTest
[stderr]     raise self.__status  # pylint: disable-msg=E0702
[stderr]   File "/usr/lib/python3.9/site-packages/avocado_vt/test.py", line 147, in setUp
[stderr]     self._runTest()
[stderr]   File "/usr/lib/python3.9/site-packages/avocado_vt/test.py", line 254, in _runTest
[stderr]     params = env_process.preprocess(self, params, env)
[stderr]   File "/usr/lib/python3.9/site-packages/virttest/error_context.py", line 135, in new_fn
[stderr]     return fn(*args, **kwargs)
[stderr]   File "/usr/lib/python3.9/site-packages/virttest/env_process.py", line 1372, in preprocess
[stderr]     if (utils_qemu.has_device_category(qemu_path, "CPU")
[stderr]   File "/usr/lib/python3.9/site-packages/virttest/utils_qemu.py", line 114, in has_device_category
[stderr]     out = _get_info(bin_path, "-device help", allow_output_check="combined")
[stderr]   File "/usr/lib/python3.9/site-packages/virttest/utils_qemu.py", line 25, in _get_info
[stderr]     return process.run(qemu_cmd, verbose=False, ignore_status=True,
[stderr] TypeError: run() got an unexpected keyword argument 'allow_output_check'
```
@beraldoleal How do you think should the virttest side be adapted?

**Comment by clebergnu** [comment]
We should have waited for a resolution on https://github.com/avocado-framework/avocado-vt/pull/3295, or, at the very least a compatibility proposal for Avocado-VT.  Anyway, let's now focus on resolving this problem.

The `allow_output_check` parameter for `avocado.utils.process.run()` was primary designed for the `--output-check-record` Avocado feature.  I don't think it's being used in Avocado-VT in the intended way, that is, the output generated will eventually be compared to a reference file to determine if the test result.

Now if it's indeed being used in an unintended way, the purpose is (if any) must be determined.  Depending on that, then a workaround or alternative solution.

To summarize it, I'm comparing the behavior with and without that argument now.  If the  outcome of this experiment confirms my guess, I'll send either a patch dropping its usage.

**Comment by clebergnu** [comment]
So, my best guess is that `allow_output_check='combined'` was being used to:

1) make sure that the logging was 100% accurate with regards to the order in which the process executed would write to stdout and stderr
2) as a simple convenience to have both stdout and stderr content in the `stdout` attribute of the `CmdResult` instance

Examples of commits with hints in the commit message include 1c465a7d855889a9ebb128f6236c8ab534a1345e and  a5c75f65513928838da5759027fde01dcd69f3f4.

For most cases then, I'd say that `allow_output_check='combined'` is nice to have, but not essential to Avocado-VT.  Also, it's being somewhat abused considering its initial purpose.

**Comment by pevogam** [comment]
Thanks for the investigation! Yes, even if this issue was reported too early for having to be resolved, it can still be used to track the particular incompatibility case. We would prefer to have issues for newly detected incompatibilities in a granular fasion rather than accumulating all of them in the shadows and having to differentiate and investigate all at the same time.

**Comment by pevogam** [comment]
Fixed in #3300.

**Comment by beraldoleal** [comment]
@clebergnu it was faster here. :) Thanks @pevogam for reporting this.
